v2x_ros_driver/include/v2x_ros_driver/v2x_radio_client.h

```diff
@@ -106,49 +104,85 @@ class V2XRadioClient
     boost::signals2::signal<void(std::vector<uint8_t> const &, uint16_t id)> onMessageReceived;
 
     /**
-     * @brief sends a udp message
+     * @brief Sends a UDP message
+     * @param message Message vector to be sent
      */
-    bool sendV2xMessage(const std::shared_ptr<std::vector<uint8_t>>&message);
+    bool sendV2xMessage(const std::shared_ptr<std::vector<uint8_t>> &message);
 
     /**
-     * @brief Validate msg_id
-     * */
+     * @brief Validate possible Message ID
+     * @param msg_id Message ID under test
+     */
     bool IsValidMsgID(const std::string &msg_id);
 
     /**
-     * @brief load list of valid msg_id
-     * */
+     * @brief Load list of valid msg_id
+     * @param fileName Configuration file to be loaded
+     */
     void loadWaveConfigIds(const std::string &fileName);
 
     /**
-     * @brief set wave config path
+     * @brief Set wave config path
+     * @param path Path where configuration file is located
     */
-   void set_wave_file_path(const std::string &path);
+    void set_wave_file_path(const std::string &path);
+
+    /**
+     * @brief Checks actual message vector size against size specified in MessageFrame
+     * @param msg_vec Full message vector under test
+     * @param start_index Starting index for detected payload within message vector
+     * @param end_index Ending index for payload within message vector
+     * @param entry Recevied data reference used to generate test vectors
+     */
+    bool isValidMsgSize(const std::vector<uint8_t> msg_vec, size_t start_index, size_t end_index, const std::vector<uint8_t> entry);
+    
+    /**
+     * @brief Checks msg_id against PSID list to make sure a WSA with valid PSID is not accidentally used instead of MessageFrame
```

Comment:
What is "WSA" in this context? Might be worth spelling it out

Reply:
Updated